adapters/repos/db/migrator.go

```diff
@@ -222,6 +222,69 @@ func (m *Migrator) RecalculateVectorDimensions(ctx context.Context) error {
 	return nil
 }
 
+func (m *Migrator) RecountProperties(ctx context.Context) error {
+	count := 0
+	m.logger.
+		WithField("action", "recount").
+		Info("Recounting properties, this may take a while")
+
+	m.db.indexLock.Lock()
+	defer m.db.indexLock.Unlock()
+	// Iterate over all indexes
+	for _, index := range m.db.indices {
+
+		// Clear the shards before counting
+		err := index.IterateShards(ctx, func(index *Index, shard *Shard) error {
+			shard.propLengths.Clear()
+			return nil
+		})
+		if err != nil {
+			m.logger.WithField("error", err).Error("could not clear prop lengths")
+		}
+
+		// Iterate over all shards
+		err = index.IterateObjects(ctx, func(index *Index, shard *Shard, object *storobj.Object) error {
+			count = count + 1
+			props, _, err := shard.analyzeObject(object)
+			if err != nil {
+				m.logger.WithField("error", err).Error("could not analyze object")
+				return nil
+			}
+
+			if err := shard.addPropLengths(props); err != nil {
+				m.logger.WithField("error", err).Error("could not add prop lengths")
+				return nil
+			}
+
+			shard.propLengths.Flush(false)
+
+			return nil
+		})
+		if err != nil {
+			m.logger.WithField("error", err).Error("could not iterate over objects")
+		}
+
+		// Flush the propLengths to disk
+		err = index.IterateShards(ctx, func(index *Index, shard *Shard) error {
+			return shard.propLengths.Flush(false)
+		})
+		if err != nil {
+			m.logger.WithField("error", err).Error("could not flush prop lengths")
+		}
+
+	}
+	go func() {
```

Comment:
It is not clear how useful is this go-routine.
It looks to like it is consuming resources to print the same message periodically.
Why not log the message in the main go-routine once?